src/gordon_gcp/clients/http.py

```diff
@@ -149,10 +149,14 @@ def __init__(self, auth_client=None, session=None):
 
                     new_req_kwargs = {
                         'params': params,
-                        'body': body,
                         'headers': headers,  # use original req headers
                         'cred_refresh_attempt': refresh_attempt + 1
                     }
+                    if req_kwargs['data']:
+                        new_req_kwargs['data'] = req_kwargs['data']
```

Comment:
That won't work because it's getting passed to request().  It needs to be body.  Honestly, I would just add "'json": json' after 152.  And probably change new_req_args to be something like recursive_args.

Reply:
I think we should scrap the recursive calling altogether. It makes things significantly more complicated for one extra retry (given that MAX_REFRESH_ATTEMPTS=2)